src/modules/mc_pos_control/MulticopterPositionControl.cpp

```diff
@@ -378,13 +375,14 @@ void MulticopterPositionControl::Run()
 			}
 
 			// limit tilt during takeoff ramupup
-			if (_takeoff.getTakeoffState() < TakeoffState::flight) {
-				_control.setTiltLimit(math::radians(_param_mpc_tiltmax_lnd.get()));
+			float tilt_limit = math::radians(_param_mpc_tiltmax_air.get());
 
-			} else {
-				_control.setTiltLimit(math::radians(_param_mpc_tiltmax_air.get()));
+			if (_takeoff.getTakeoffState() < TakeoffState::flight) {
+				tilt_limit = math::radians(_param_mpc_tiltmax_lnd.get());
 			}
 
+			_control.setTiltLimit(_tilt_limit_slew_rate.update(tilt_limit, dt));
+
```

Comment:
I'm not convinced this is better than before; it creates a new (non const) variable with a really large scope.

Reply:
maybe
```
const float tilt_limit_deg = (_takeoff.getTakeoffState() < TakeoffState::flight)
                           ? _param_mpc_tiltmax_lnd.get()
                           : _param_mpc_tiltmax_air.get();
_control.setTiltLimit(_tilt_limit_slew_rate.update(math::radians(tilt_limit_deg), dt));

Reply:
Yeah, personal preference. I agree your suggestion is more readable even though I don't like the ternary operator.

Reply:
Done in 533dc43